### DISCUSSION TITLE
How merge without Pull Request?

### DISCUSSION THREAD

**Discussion by andr1972** [community/community#22354] [New to GitHub]
<p>If I want merge master&lt;-branch, both direction will be merged or master-&gt;branch if pull will not closed.</p>

**Comment by Wabri** [comment, marked as answer]
<p>I do not know if I understand the question, but I try to answer…</p>
<p>For the question of the title you can merge branch without pull request in your local machine, like the help page says:</p>
<pre><code>git merge --help
</code></pre>
<p>the output is:</p>
<pre><code>[...]

Assume the following history exists and the current branch is "master":

                     A---B---C topic
                    /
               D---E---F---G

Then "git merge topic" will replay the changes made on the topic branch
since it diverged from master (i.e., E) until its current commit (C) on
top of master, and record the result in a new commit along with the names
of the two parent commits and a log message from the user describing the changes.

                     A---B---C topic
                    / \
               D---E---F---G---H master

[...]
</code></pre>
<p>Accordingly to merge the branch named “branch” into branch “master” you need to be in the branch “master” and than run the command:</p>
<pre><code>git merge branch
</code></pre>
<p>now all the changes made on “branch” are merged into master, but the “branch” are still there and will continue to exists until you not delete it with:</p>
<pre><code>git branch -D branch
</code></pre>
<p>If you want master merge in branch you can reverse the commands above:</p>
<ol>
<li>
<p> move to “branch”:</p>
<p>git checkout branch</p>
</li>
<li>
<p>merge “master” in “branch”:</p>
<p>git merge master</p>
</li>
</ol>
<p>Done.</p>
<p>Hope to solve your problem :slightly_smiling_face:</p>
<p>-Gabriele-</p>
